nwb_conversion_tools/basesortingextractorinterface.py

```diff
@@ -71,13 +55,14 @@ def run_conversion(self, nwbfile: NWBFile, metadata: dict, stub_test: bool = Fal
             sorting_extractor = self.sorting_extractor
 
         for metadata_column in metadata['UnitProperties']:
-            property_descriptions.update({metadata_column['name']: metadata_column['description']})
-            for unit_id in sorting_extractor.get_unit_ids():
-                if metadata_column['name'] == 'electrode_group':
-                    data = nwbfile.electrode_groups[metadata_column['data'][unit_id]]
-                else:
-                    data = metadata_column['data'][unit_id]
-                sorting_extractor.set_unit_property(unit_id, metadata_column['name'], data)
+            if len(metadata_column['data']) == len(sorting_extractor.get_unit_ids()):
```

Comment:
This change block looks messy, but it's only the addition of this if condition. Basically, if there is disagreement between the dimensionality of processed spiking unit metadata (like cell type) and what is directly pulled out of the .res and .clu files, the metadata won't be added.